clang/SourceLocation.d

```diff
@@ -47,4 +56,15 @@ struct SourceLocation
         auto s = spelling;
         return format("SourceLocation(file = %s, line = %d, column = %d, offset = %d)", s.file, s.line, s.column, s.offset);
     }
+
+    static bool lexicalLess(in SourceLocation a, in SourceLocation b)
```

Comment:
It is used as predicate in some another place, I prefer predicated to be binary functions, but I suppose I can make a symmetry and make it a usual method.